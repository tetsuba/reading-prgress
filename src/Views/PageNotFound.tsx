import Button from '../Components/Button/Button'
import { useNavigate } from 'react-router-dom'

export default function PageNotFound() {
    const navigate = useNavigate()
    return (
        <main className="grid min-h-full place-items-center bg-white py-24 px-6 sm:py-32 lg:px-8">
            <div className="text-center">
                <p className="text-base font-semibold text-indigo-600">404</p>
                <h1 className="mt-4 text-3xl font-bold tracking-tight text-gray-900 sm:text-5xl">
                    Page not found
                </h1>
                <p className="mt-6 text-base leading-7 text-gray-600">
                    Sorry, we couldn’t find the page you’re looking for.
                </p>
                <div className="mt-10 flex items-center justify-center gap-x-6">
                    <Button
                        dataTestid="go-back-button"
                        type="button"
                        template="primary"
                        clickHandler={() => navigate(-1)}
                    >
                        Go back
                    </Button>
                </div>
            </div>
        </main>
    )
}
