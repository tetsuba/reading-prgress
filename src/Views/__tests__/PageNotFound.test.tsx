import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import PageNotFound from '../PageNotFound'

const mockNavigate = vi.fn()
vi.mock('react-router-dom', async () => {
    const actual = await vi.importActual('react-router-dom')
    return {
        ...(actual as object),
        useNavigate: () => mockNavigate
    }
})

describe('PageNotFound', () => {
    test('render page and click on go back button', async () => {
        const { asFragment } = render(<PageNotFound />)
        expect(asFragment()).toMatchSnapshot()
        fireEvent.click(screen.getByTestId('go-back-button'))
        await waitFor(() => expect(mockNavigate).toHaveBeenCalled())
    })
})
