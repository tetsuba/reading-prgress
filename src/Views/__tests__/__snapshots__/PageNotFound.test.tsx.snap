// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageNotFound > render page and click on go back button 1`] = `
<DocumentFragment>
  <main
    class="grid min-h-full place-items-center bg-white px-6 py-24 sm:py-32 lg:px-8"
  >
    <div
      class="text-center"
    >
      <p
        class="text-base font-semibold text-indigo-600"
      >
        404
      </p>
      <h1
        class="mt-4 text-3xl font-bold tracking-tight text-gray-900 sm:text-5xl"
      >
        Page not found
      </h1>
      <p
        class="mt-6 text-base leading-7 text-gray-600"
      >
        Sorry, we couldn’t find the page you’re looking for.
      </p>
      <div
        class="mt-10 flex items-center justify-center gap-x-6"
      >
        <button
          class="rounded-lg font-medium text-base text-center px-5 py-2.5 text-white bg-blue-700 hover:bg-blue-800 "
          data-testid="go-back-button"
          template="primary"
        >
          Go back
        </button>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
